corehq/apps/users/models.py

```diff
@@ -3041,3 +3041,26 @@ def role(self):
         elif self.domain:
             return CouchUser.from_django_user(self.user).get_domain_membership(self.domain).role
         return None
+
+
+class UserHistory(models.Model):
+    """
+    HQ Adaptation of Django's LogEntry model
+    """
+    CREATE = 1
+    UPDATE = 2
+    DELETE = 3
+
+    ACTION_CHOICES = (
+        (CREATE, _('Create')),
+        (UPDATE, _('Update')),
+        (DELETE, _('Delete')),
+    )
+    domain = models.CharField(max_length=255, null=True)
+    user_type = models.CharField(max_length=255)
+    user_id = models.CharField(max_length=128)
+    by_user_id = models.CharField(max_length=128)
```

Comment:
agreed about needing a moment to understand which is which. I am open to suggestions on this.
This is the user that created/updated/deleted the user object, so just `created_by` won't suffice. Can think of a longer name like `changed_by_user_id` or just `changed_by`, which could be useful.

Reply:
I like `changed_by`. Should `created_on` also become `changed_on`?

Reply:
you mean `created_at`? That is meant to say what time the log was created at.

(Would update to use `changed_by` once all child PRs are merged for easier change)

Reply:
I think the user and timestamp should be named similarly. If `created_at` is to signify the time the log was created, then I'd use `created_by` to signify the user who created the log. If `changed_by/changed_on` sounds better, that's fine too.

Reply:
thank you.

Updated. Used "changed_at" instead since its time and not date, so "at" sounds right.

https://github.com/dimagi/commcare-hq/pull/29887/commits/548ef72bb3807723a3ea618e95b35fe0429edb47